packages/components/src/font-size-picker/index.js

```diff
@@ -26,84 +38,64 @@ function FontSizePicker( {
 	value,
 	withSlider = false,
 } ) {
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+	const [ currentSelectValue, setCurrentSelectValue ] = useState( getInitialSelectValue( fontSizes, value ) );
+
 	if ( disableCustomFontSizes && ! fontSizes.length ) {
 		return null;
 	}
+
 	const onChangeValue = ( event ) => {
 		const newValue = event.target.value;
+		setCurrentSelectValue( getInitialSelectValue( fontSizes, Number( newValue ) ) );
```

Comment:
With `getInitialSelectValue` being used here in `onChangeValue` as well, its name no longer makes as much sense—it's no longer being used for just the initial value. Maybe something like `getSelectValueFromFontSize` could work better.

Reply:
Done!